adr/2025-06-03-integrating-the-language-pack-into-platform.md

```diff
@@ -0,0 +1,49 @@
+
+1. Translations are updated in CrowdIn (by the community or shopware)
+2. Merging the changes from CrowdIn into `shopware/translations` (automated, may require manual review)
+3. Merging the changes from `shopware/translations` into `shopware/SwagLanguagePack`, which primarily just distributes json files (automated, may require manual review)
+4. Publishing Language Pack releases the store (automated, may require manual input)
+5. Updating Language Pack installation
+   - OnPrem: Manually updating the plugin in a Shopware installation
+   - Cloud: Updating the docker image dependency
+
+This sequence also introduces overhead in maintenance and CI/CD
+
+### Background and Motivation
+* The current setup was influenced by limitations of our previous GitLab-based workflow, where distribution via plugin was the most practical method. This workflow has been in place for ~6 years.
+* Crowdin remains our source of truth for all supported language snippets, and community members can contribute translations directly.
+* `shopware/translations` acts as an intermediary data source, so we are not coupled to Crowdin in core Shopware repositories.
+* The primary goal of this change is to reduce maintenance effort by replacing steps 3–5 with a single step, and removing `shopware/SwagLanguagePack` from the workflow entirely. The new step should be in `shopware/shopware`.
+
+## Decision
+We will implement a new service in `shopware/shopware` (ie as part of the Shopware platform) to download translations right from their [GitHub Repository](https://github.com/shopware/translations/) and manage them without the need to install any extension.
+Translations will be downloaded as JSON files (via admin user interaction or command execution) and stored on the local file system, just like existing platform snippet files.
+In addition, we will provide new bin/console commands in `shopware/shopware` to manage installed languages, for example when building an image for deployment. The initial set of commands will look like:
+
+```bash
+$ php bin/console language
+
+Available commands:
+    install [languages] [--all]
+    activate [languages] [--all]
+    deactivate [languages] [--all]
+    uninstall [languages] [--all]
+    list
+```
+
+## Consequences
+* The Language Pack plugin will be maintained for Shopware versions < v6.8.0.
+* Translations can be installed and updated on-demand, instead of waiting for platform/plugin release cycles.
+* Translation versions will be mapped to platform version ranges.
+* The general translations workflow remains the same. This has no impact on other extensions and their snippet files.
+* For admin users the UX will improve: Available languages will now be listed directly in the Languages module and can be installed with a single click.
+* In the future this will allow us to improve the first run wizard UX, for example by preconfiguring countries and currencies based on a selected language.
```

Comment:
I thought that would be an independent change 🤔 Why mention it here?

Reply:
do not confuse an ADR with an epic :wink:

Reply:
I will delete this point 👍